arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -3147,44 +3149,60 @@ std::unique_ptr<AqlIndexStreamIterator> RocksDBVPackIndex::streamForCondition(
   return stream;
 }
 
-bool RocksDBVPackIndex::checkSupportsStreamInterface(
+Index::StreamSupportResult RocksDBVPackIndex::checkSupportsStreamInterface(
     std::vector<std::vector<basics::AttributeName>> const& coveredFields,
-    IndexStreamOptions const& streamOpts) noexcept {
-  // TODO expand this for fixed values that can be moved into the index
-
+    std::vector<std::vector<basics::AttributeName>> const& fields,
+    bool isUnique, IndexStreamOptions const& streamOpts) noexcept {
   // we can only project values that are in range
   for (auto idx : streamOpts.projectedFields) {
     if (idx > coveredFields.size()) {
-      return false;
+      return StreamSupportResult::makeUnsupported();
     }
   }
 
-  // for persisted indexes, we can only use a prefix of the indexed keys
-  std::size_t idx = 0;
+  // only one key fields is supported currently
   if (streamOpts.usedKeyFields.size() != 1) {
-    return false;
+    return StreamSupportResult::makeUnsupported();
+  }
+  if (streamOpts.usedKeyFields[0] >= fields.size()) {
+    return StreamSupportResult::makeUnsupported();
   }
```

Comment:
This condition is new and I don't really get why this is needed?

Reply:
It was missing. We can't have a join key on a stored value.